src/libs/Navigation/NavigationRoot.js

```diff
@@ -1,115 +1,52 @@
 import _ from 'underscore';
 import React, {Component} from 'react';
-import {ActivityIndicator, Linking, View} from 'react-native';
 import PropTypes from 'prop-types';
-import {
-    getStateFromPath,
-    getPathFromState,
-    NavigationContainer,
-} from '@react-navigation/native';
-import {withOnyx} from 'react-native-onyx';
+import {getPathFromState, NavigationContainer} from '@react-navigation/native';
 import {navigationRef} from './Navigation';
 import linkingConfig from './linkingConfig';
 import AppNavigator from './AppNavigator';
-import getPathName from './getPathName';
-import ONYXKEYS from '../../ONYXKEYS';
 import ROUTES from '../../ROUTES';
-import styles from '../../styles/styles';
-import themeColors from '../../styles/themes/default';
 import {updateCurrentlyViewedReportID} from '../actions/Report';
 import {setCurrentURL} from '../actions/App';
+import FullScreenLoadingIndicator from '../../components/FullscreenLoadingIndicator';
 
 const propTypes = {
     // Whether the current user is logged in with an authToken
     authenticated: PropTypes.bool.isRequired,
-
-    // The current reportID that we are navigated to or should show in the ReportScreen
-    currentlyViewedReportID: PropTypes.string,
-};
-
-const defaultProps = {
-    currentlyViewedReportID: null,
 };
 
 class NavigationRoot extends Component {
     constructor(props) {
         super(props);
 
-        this.state = {
-            loading: true,
-            initialState: undefined,
-        };
+        this.handleStateChange = this.handleStateChange.bind(this);
     }
 
-    componentDidMount() {
-        Linking.getInitialURL()
-            .then((initialUrl) => {
-                // On web we should be able to parse this. It will be null on native for now until deep links are
-                // hooked up
-                const path = getPathName(initialUrl);
-                let initialState = getStateFromPath(path, linkingConfig.config);
-                setCurrentURL(path);
```

Comment:
The entire logic here, seems no longer necessary - `setCurrentURL` is covered by the call in 
 `onStateChange`. 
- initial state is resolved correctly
- deep links are still resolved. E.g. try `localhost:8080/setpassword/10024706/EHOHMPGWM`